compiler/src/iree/compiler/Codegen/Common/ReconcileTranslationInfo.cpp

```diff
@@ -362,77 +364,112 @@ void ReconcileTranslationInfoPass::runOnOperation() {
-      reconcileWorkgroupSize(translationInfos);
-  if (failed(reconciledWorkgroupSize)) {
-    exportOp.emitOpError("failed to reconcile workgroup sizes");
-    return signalPassFailure();
-  }
-  if (reconciledWorkgroupSize->size() > 3) {
-    exportOp.emitOpError(
-        "reconciled workgroup size is greater than 3 (illegal)");
-    return signalPassFailure();
-  }
-  std::array<int64_t, 3> workgroupSize = {1, 1, 1};
-  for (auto [index, size] : llvm::enumerate(reconciledWorkgroupSize.value())) {
-    workgroupSize[index] = size;
-  }
-  auto workgroupSizeArrayAttr = rewriter.getIndexArrayAttr(workgroupSize);
-  exportOp.setWorkgroupSizeAttr(workgroupSizeArrayAttr);
-
-  // Reconcile subgroup sizes.
-  FailureOr<int64_t> reconciledSubgroupSize =
-      reconcileSubgroupSize(translationInfos);
-  if (failed(reconciledSubgroupSize)) {
-    exportOp.emitOpError("failed to reconcile subgroup size");
-    return signalPassFailure();
-  }
-  if (reconciledSubgroupSize.value() != int64_t()) {
-    exportOp.setSubgroupSizeAttr(
-        rewriter.getIndexAttr(reconciledSubgroupSize.value()));
+      auto translationInfo = getTranslationInfo(funcOp);
+      if (!translationInfo) {
+        // No translation info means nothing to reconcile.
+        continue;
+      }
+      translationInfos.push_back(translationInfo);
+
+      // The following is moving the target-func-attrs specification from
+      // translation info into the func-like op. This is not the best
+      // place to do this, but the intent is after this pass all the
+      // lowering configs and translation infos will be deleted.
+      DictionaryAttr targetFuncAttrs = getTargetFuncAttrs(translationInfo);
+      if (targetFuncAttrs) {
+        funcOp->setAttr("llvm_func_attrs", targetFuncAttrs);
+      }
+    }
+
+    // Reconcile workgroup sizes.
+    FailureOr<SmallVector<int64_t>> reconciledWorkgroupSize =
+        reconcileWorkgroupSize(translationInfos);
+    if (failed(reconciledWorkgroupSize)) {
+      exportOp.emitOpError("failed to reconcile workgroup sizes");
+      return signalPassFailure();
+    }
+    if (reconciledWorkgroupSize->size() > 3) {
+      exportOp.emitOpError(
+          "reconciled workgroup size is greater than 3 (illegal)");
+      return signalPassFailure();
+    }
+    std::array<int64_t, 3> workgroupSize = {1, 1, 1};
+    for (auto [index, size] :
+         llvm::enumerate(reconciledWorkgroupSize.value())) {
+      workgroupSize[index] = size;
+    }
+    auto workgroupSizeArrayAttr = rewriter.getIndexArrayAttr(workgroupSize);
+    exportOp.setWorkgroupSizeAttr(workgroupSizeArrayAttr);
+
+    // Reconcile subgroup sizes.
+    FailureOr<int64_t> reconciledSubgroupSize =
+        reconcileSubgroupSize(translationInfos);
+    if (failed(reconciledSubgroupSize)) {
+      exportOp.emitOpError("failed to reconcile subgroup size");
+      return signalPassFailure();
+    }
+    if (reconciledSubgroupSize.value() != int64_t()) {
```

Comment:
I vaguely remember that I tried to convinced Mahesh or others to replace the `int64_t()` with `0`, which is more straightforward to me, but I failed. I don't remember the reason, and I can't find the old discussion.

Let me try it again, can we use `0`? It looks more straightforward to me.

Reply:
0 makes sense to me! I went ahead and changed the other instance of it in this file too.